torch/nested/_internal/nested_tensor.py

```diff
@@ -252,18 +260,14 @@ def __tensor_unflatten__(inner_tensors: Dict, meta, outer_size, outer_stride):
             metadata_cache["min_seqlen"] = min_seqlen_tensor
         if max_seqlen_tensor is not None:
             metadata_cache["max_seqlen"] = max_seqlen_tensor
-
         ragged_idx = meta["ragged_idx"]
 
-        # Note that we cannot simply check if is_fake(values) because
-        # during aot autograd, FunctionalTensors are not fake but hold
-        # symbolic sizes.
+        # Alternatively, we could make it the caller's responsibility to
+        # cache it. But this heuristic seems simple enough.
         ragged_source = offsets if lengths is None else lengths
-        if has_free_symbols(ragged_source) or has_free_symbols(values):
-            # Associate offsets or lengths (possibly fake, possibly functionalized)
-            # with the ragged_size.
+        if isinstance(ragged_source, FakeTensor):
             ragged_size = outer_size[ragged_idx]
-            _tensor_symint_registry[ragged_source] = ragged_size
+            ragged_source.nested_int_memo = ragged_size
```

Comment:
I think `ragged_source can be a `FunctionalTensor(FakeTensor)` here as well. Is it intentional that we are only setting the memo in the FakeTensor case?

(just thinking out loud - I think the answer is... yes, because you've written code to make `FunctionalTensor.nested_into_memo` a pass-through to grab the nested_int from the underlying FakeTensor. But might be worth calling out the invariant in a comment somewhere, that only the `FakeTensor` offsets gets a memo)

Reply:
Yes, will add a comment

Reply:
This particular piece of code actually goes away (see next PR), so I added the comment to somewhere else (get_nested_int logic).